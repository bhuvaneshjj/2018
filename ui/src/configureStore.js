import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import appReducer from './App/appReducer';
import { websocketConnected, websocketDisconnected } from './App/AppActions';
import { setHighestMezicasId } from './casomeric/Mezicasy/MezicasyActions';

const demoStartujiciState = {
  startujici: [
    { id: 0, cislo: 7, dokonceno: null },
    { id: 1, cislo: 4, dokonceno: null },
    { id: 17, cislo: 16, dokonceno: null },
    { id: 2, cislo: 25, dokonceno: false },
    { id: 3, cislo: 9, dokonceno: null },
    { id: 12, cislo: 15, dokonceno: null },
    { id: 9, cislo: 1, dokonceno: false },
    { id: 29, cislo: 8, dokonceno: null },
    { id: 5, cislo: 59, dokonceno: null },
    { id: 11, cislo: 43, dokonceno: null },
    { id: 18, cislo: 42, dokonceno: false },
    { id: 10, cislo: 33, dokonceno: null },
    { id: 7, cislo: 21, dokonceno: false },
    { id: 13, cislo: 24, dokonceno: null },
    { id: 34, cislo: 27, dokonceno: null },
    { id: 14, cislo: 22, dokonceno: null },
    { id: 58, cislo: 30, dokonceno: null },
    { id: 15, cislo: 23, dokonceno: null },
    { id: 59, cislo: 26, dokonceno: null }
  ]
};

const loadState = () => {
  try {
    const serializedState = localStorage.getItem('state');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    console.log('Problem while loading app state from the local storage', err);
    return undefined;
  }
};

const saveState = state => {
  try {
    const serializedState = JSON.stringify(state);
    localStorage.setItem('state', serializedState);
  } catch (err) {
    console.log('Problem while storing app state to the local storage', err);
  }
};

const setupWsClient = (wsClient, store) => {
  wsClient.setCallbacks({
    onConnect: () => {
      store.dispatch(websocketConnected());
    },
    onClose: () => store.dispatch(websocketDisconnected())
  });

  try {
    /* Start asynchronous connect to the websocket server.
     WsClient will retry indefinitely if the connection cannot be established. */
    wsClient.connect();
  } catch (err) {
    // Silently ignore any errors. They should have been dispatched from WsClient anyway.
  }
};

const configureStore = (wsClient, initialStateParam = loadState()) => {
  const initialState = initialStateParam || {};
  initialState.startujici = demoStartujiciState.startujici;

  if (initialState.casomeric && initialState.casomeric.mezicasy) {
    let highestId = 0;

    initialState.casomeric.mezicasy.forEach(mezicas => {
      if (mezicas.id > highestId) {
        highestId = mezicas.id;
      }
    });

    setHighestMezicasId(highestId);
  }

  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(
    appReducer,
    initialState,
    composeEnhancers(applyMiddleware(thunk.withExtraArgument(wsClient)))
  );

  store.subscribe(() => {
    const state = store.getState();
    saveState({
      auth: {
        authenticated: state.auth.authenticated,
        decodedToken: state.auth.decodedToken,
        token: state.auth.token
      },
      casomeric: { stopky: state.casomeric.stopky, mezicasy: state.casomeric.mezicasy }
    });
  });

  if (wsClient) {
    setupWsClient(wsClient, store);
  }

  return store;
};

export default configureStore;
