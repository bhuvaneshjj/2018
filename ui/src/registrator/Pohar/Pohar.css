.Pohar__div {
  padding-left: 30px;
  padding-right: 50px;
}

.Pohar__buttons {
  margin-left: 1em;
}

.Pohar__button {
  margin-left: 5px;
}
