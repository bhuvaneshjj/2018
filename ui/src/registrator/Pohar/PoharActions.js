export const narokovaneFilterChange = () => ({ type: 'POHAR_NAROKOVANE_FILTER_CHANGE' });

export const neprevzateFilterChange = () => ({ type: 'POHAR_NEPREVZATE_FILTER_CHANGE' });
