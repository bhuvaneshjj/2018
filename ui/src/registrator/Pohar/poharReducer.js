import { UCASTI_NA_POHAR } from '../../common';
import { AKTUALNI_ROK } from '../../constants';
import { sortForColumn } from '../../sort';
import {
  createFilterableReducer,
  initialState as filterableInitialState
} from '../Filterable/filterableReducer';
import {
  createUcastniciTableReducer,
  initialState as ucastniciTableInitialState
} from '../UcastniciTable/ucastniciTableReducer';

export const initialState = {
  narokovaneFilter: false, // jen potenciální nárok z přihlášky
  neprevzateFilter: false, // jen nepřevzaté
  ...filterableInitialState,
  ...ucastniciTableInitialState
};

const filterableReducer = createFilterableReducer('POHAR');
const ucastniciTableReducer = createUcastniciTableReducer('POHAR');
const poharReducer = (state = initialState, action) => {
  state = filterableReducer(state, action); // eslint-disable-line no-param-reassign
  state = ucastniciTableReducer(state, action); // eslint-disable-line no-param-reassign

  switch (action.type) {
    case 'POHAR_NAROKOVANE_FILTER_CHANGE':
      return { ...state, narokovaneFilter: !state.narokovaneFilter };
    case 'POHAR_NEPREVZATE_FILTER_CHANGE':
      return { ...state, neprevzateFilter: !state.neprevzateFilter };
    default:
      return state;
  }
};

export default poharReducer;

export const getPoharySorted = ({
  narokovaneFilter,
  neprevzateFilter,
  textFilter,
  sortColumn,
  sortDir,
  kategorie,
  ucastnici
}) => {
  const maratonci = ucastnici.allIds
    .map(id => {
      const ucastnik = ucastnici.byIds[id];
      const dokoncene = ucastnik.roky
        .map(rok => {
          const { prihlaska, vykon } = ucastnik[rok];
          const { kategorie: kategorieId } = prihlaska;
          return kategorie[kategorieId].typ === 'maraton' && vykon && vykon.dokonceno === true
            ? rok
            : undefined;
        })
        .filter(rok => rok !== undefined);

      const prihlaseno = ucastnik.roky.reduce((result, rok) => {
        const { prihlaska } = ucastnik[rok];
        const { kategorie: kategorieId } = prihlaska;
        return result || (kategorie[kategorieId].typ === 'maraton' && rok === AKTUALNI_ROK);
      }, false);

      const { prijmeni, jmeno, narozeni } = ucastnik[ucastnik.roky[0]].udaje;
      const predano = (ucastnik.pohar && ucastnik.pohar.predano) || 0;
      const neprevzato = Math.floor(dokoncene.length / UCASTI_NA_POHAR || 0) - predano;
      const narok = prihlaseno && (dokoncene.length + 1) % UCASTI_NA_POHAR === 0;
      return {
        id,
        prijmeni,
        jmeno,
        narozeni,
        pohary: { narok, predano, neprevzato },
        ucasti: { dokoncene, prihlaseno }
      };
    })
    .filter(({ ucasti: { dokoncene, prihlaseno } }) => dokoncene.length > 0 || prihlaseno);

  const afterTextFilter = maratonci.filter(
    ({ jmeno, prijmeni }) =>
      prijmeni.toLowerCase().startsWith(textFilter) || jmeno.toLowerCase().startsWith(textFilter)
  );

  const afterNarokovaneFilter = narokovaneFilter
    ? afterTextFilter.filter(({ pohary: { narok } }) => narok > 0)
    : afterTextFilter;

  const afterNeprevzateFilter = neprevzateFilter
    ? afterNarokovaneFilter.filter(({ pohary: { neprevzato } }) => neprevzato > 0)
    : afterNarokovaneFilter;

  return sortForColumn({ data: afterNeprevzateFilter, sortColumn, sortDir });
};
