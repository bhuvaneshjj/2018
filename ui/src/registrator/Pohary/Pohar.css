.Pohar {
  margin-left: 5px;
}
