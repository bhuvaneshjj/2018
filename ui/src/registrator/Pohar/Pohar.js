import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Button, Glyphicon } from 'react-bootstrap';
import { narozeniToStr } from '../../Util';
import TextFilter from '../Filterable/TextFilter';
import Zobrazeno from '../Filterable/Zobrazeno';
import UcastniciTableContainer from '../UcastniciTable/UcastniciTableContainer';
import './Pohar.css';

const narozeniFormat = ({ cellData }) => narozeniToStr(cellData);

const prijmeniFormat = args => (
  <Link to={`/prihlasky/${args.data[args.rowIndex].id}`}>{args.cellData}</Link>
);

const Pohar = ({
  actionPrefix,
  narokovaneFilter,
  neprevzateFilter,
  pohary,
  reduxName,
  textFilter,
  onNarokovaneFilterChange,
  onNeprevzateFilterChange,
  onTextFilterChange
}) => {
  const columns = [
    {
      cellClassNames: () => ['align-left'],
      cellDataFormatter: prijmeniFormat,
      key: 'prijmeni',
      label: 'příjmení',
      sortable: true,
      width: 100
    },
    {
      cellClassNames: () => ['align-left'],
      key: 'jmeno',
      label: 'jméno',
      sortable: true,
      width: 90
    },
    {
      cellClassNames: () => ['align-right'],
      cellDataFormatter: narozeniFormat,
      key: 'narozeni',
      label: 'narození',
      sortable: true,
      width: 100
    }
  ];

  return (
    <div className="Pohar__div UcastniciTable_container">
      <div>
        <TextFilter filter={textFilter} onChange={onTextFilterChange} />

        <span className="Pohar__buttons Bootstrap-buttons--active">
          <Button
            active={narokovaneFilter}
            bsStyle="success"
            className="Pohar__button"
            onClick={onNarokovaneFilterChange}
          >
            <Glyphicon glyph="download" /> Nárok na pohár
          </Button>
          <Button
            active={neprevzateFilter}
            bsStyle="info"
            className="Pohar__button"
            onClick={onNeprevzateFilterChange}
          >
            <Glyphicon glyph="upload" /> Nepřevzatý pohár
          </Button>
        </span>

        <Zobrazeno numberOfItems={pohary.length} />
      </div>

      <UcastniciTableContainer
        actionPrefix={actionPrefix}
        columns={columns}
        data={pohary}
        fixedColumnCount={3}
        reduxName={reduxName}
        rowHeight={35}
      />
    </div>
  );
};

Pohar.propTypes = {
  actionPrefix: PropTypes.string.isRequired,
  narokovaneFilter: PropTypes.bool.isRequired,
  neprevzateFilter: PropTypes.bool.isRequired,
  pohary: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      prijmeni: PropTypes.string.isRequired,
      jmeno: PropTypes.string.isRequired,
      narozeni: PropTypes.shape({
        den: PropTypes.number,
        mesic: PropTypes.number,
        rok: PropTypes.number.isRequired
      }).isRequired
    }).isRequired
  ).isRequired,
  reduxName: PropTypes.string.isRequired,
  textFilter: PropTypes.string.isRequired,
  onNarokovaneFilterChange: PropTypes.func.isRequired,
  onNeprevzateFilterChange: PropTypes.func.isRequired,
  onTextFilterChange: PropTypes.func.isRequired
};

export default Pohar;
