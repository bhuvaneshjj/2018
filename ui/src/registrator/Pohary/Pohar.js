import React from 'react';
import PropTypes from 'prop-types';
import './Pohar.css';
import pohar from './pohar.svg';

const Pohar = ({ sizePercentage }) => (
  <img
    src={pohar}
    alt="pohár"
    className="Pohar"
    title="pohár"
    width={(sizePercentage * 517) / 100}
    height={(sizePercentage * 643) / 100}
  />
);

Pohar.propTypes = {
  sizePercentage: PropTypes.number.isRequired
};

Pohar.defaultProps = {
  sizePercentage: 100
};

export default Pohar;
