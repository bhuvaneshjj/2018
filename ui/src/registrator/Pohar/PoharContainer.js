import { connect } from 'react-redux';
import { createTextFilterChange } from '../Filterable/FilterableActions';
import { narokovaneFilterChange, neprevzateFilterChange } from './PoharActions';
import { getPoharySorted } from './poharReducer';
import Pohar from './Pohar';

// Don't forget to update areStatesEqual!
const mapStateToProps = ({ entities, registrator: { pohar } }) => {
  const { narokovaneFilter, neprevzateFilter, textFilter } = pohar;

  return {
    actionPrefix: 'POHAR',
    narokovaneFilter,
    neprevzateFilter,
    pohary: getPoharySorted({ ...entities, ...pohar }),
    reduxName: 'pohar',
    textFilter
  };
};

const mapDispatchToProps = dispatch => ({
  onNarokovaneFilterChange: () => dispatch(narokovaneFilterChange()),
  onNeprevzateFilterChange: () => dispatch(neprevzateFilterChange()),
  onTextFilterChange: text => dispatch(createTextFilterChange('POHAR')(text))
});

const areStatesEqual = (next, prev) =>
  prev.entities === next.entities && prev.registrator.pohar === next.registrator.pohar;

const PoharContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
  null,
  {
    areStatesEqual
  }
)(Pohar);

export default PoharContainer;
