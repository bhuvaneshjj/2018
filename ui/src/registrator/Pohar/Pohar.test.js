import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Pohar from './Pohar';

const mockStore = configureStore();

const state = { registrator: { pohar: {} } };
const store = mockStore(state);
store.dispatch = jest.fn();

const commonProps = {
  actionPrefix: 'POHAR',
  narokovaneFilter: false,
  neprevzateFilter: false,
  reduxName: 'pohar',
  textFilter: '',
  onNarokovaneFilterChange: jest.fn(),
  onNeprevzateFilterChange: jest.fn(),
  onTextFilterChange: jest.fn()
};

const pohary = [
  {
    id: '5a09b1fd371dec1e99b7e1c9',
    prijmeni: 'Balabák',
    jmeno: 'Roman',
    narozeni: { rok: 1956 }
  },
  {
    id: '6f09b1fd371dec1e99b7e1c9',
    prijmeni: 'Sukdoláková',
    jmeno: 'Martina',
    narozeni: { rok: 1963, mesic: 12, den: 7 },
    obec: 'Zlín'
  },
  {
    id: '7a09b1fd371dec1e99b7e142',
    prijmeni: 'Zralá',
    jmeno: 'Hana',
    narozeni: { den: 25, mesic: 7, rok: 1999 }
  }
];

it('žádný pohár', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Pohar pohary={[]} {...commonProps} />
      </MemoryRouter>
    </Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

it('dva pohároví', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Pohar pohary={pohary} {...commonProps} />
      </MemoryRouter>
    </Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
