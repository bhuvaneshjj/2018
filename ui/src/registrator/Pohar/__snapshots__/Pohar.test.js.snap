// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dva pohároví 1`] = `
<div
  className="Pohar__div UcastniciTable_container"
>
  <div>
    <input
      className="TextFilter__input"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Filtr na příjmení a jméno"
      type="text"
      value=""
    />
    <span
      className="Pohar__buttons Bootstrap-buttons--active"
    >
      <button
        className="Pohar__button btn btn-success"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          className="glyphicon glyphicon-download"
        />
         Nárok na pohár
      </button>
      <button
        className="Pohar__button btn btn-info"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          className="glyphicon glyphicon-upload"
        />
         Nepřevzatý pohár
      </button>
    </span>
    <span
      className="Zobrazeno"
    >
      zobrazeno: 
      <span
        className="badge"
      >
        3
      </span>
    </span>
  </div>
  <div
    className="withResponsive"
  >
    <div
      className="UcastniciTable_GridRow"
    >
      <div
        className="UcastniciTable_LeftSideGridContainer"
        style={
          Object {
            "flexBasis": "290px",
            "top": 0,
          }
        }
      >
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid UcastniciTable_HeaderGrid"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 35,
              "overflowX": "hidden",
              "overflowY": "hidden",
              "position": "relative",
              "width": 290,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        >
          <div
            className="ReactVirtualized__Grid__innerScrollContainer"
            role="rowgroup"
            style={
              Object {
                "height": 35,
                "maxHeight": 35,
                "maxWidth": 290,
                "overflow": "hidden",
                "pointerEvents": "",
                "position": "relative",
                "width": 290,
              }
            }
          >
            <div
              className="UcastniciTable_headerCell"
              style={
                Object {
                  "height": 35,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 100,
                }
              }
            >
              <button
                className="SortHeader"
                onClick={[Function]}
              >
                příjmení
                 
                
              </button>
            </div>
            <div
              className="UcastniciTable_headerCell"
              style={
                Object {
                  "height": 35,
                  "left": 100,
                  "position": "absolute",
                  "top": 0,
                  "width": 90,
                }
              }
            >
              <button
                className="SortHeader"
                onClick={[Function]}
              >
                jméno
                 
                
              </button>
            </div>
            <div
              className="UcastniciTable_headerCell"
              style={
                Object {
                  "height": 35,
                  "left": 190,
                  "position": "absolute",
                  "top": 0,
                  "width": 100,
                }
              }
            >
              <button
                className="SortHeader"
                onClick={[Function]}
              >
                narození
                 
                
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="UcastniciTable_LeftSideGridContainer"
        style={
          Object {
            "flexBasis": "290px",
            "top": 35,
          }
        }
      >
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid UcastniciTable_LeftSideGrid"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 105,
              "overflowX": "hidden",
              "overflowY": "hidden",
              "position": "relative",
              "width": 290,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        >
          <div
            className="ReactVirtualized__Grid__innerScrollContainer"
            role="rowgroup"
            style={
              Object {
                "height": 105,
                "maxHeight": 105,
                "maxWidth": 290,
                "overflow": "hidden",
                "pointerEvents": "",
                "position": "relative",
                "width": 290,
              }
            }
          >
            <div
              className="align-left UcastniciTable_oddRow UcastniciTable_cell"
              style={
                Object {
                  "height": 35,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 100,
                }
              }
            >
              <a
                href="/prihlasky/5a09b1fd371dec1e99b7e1c9"
                onClick={[Function]}
              >
                Balabák
              </a>
            </div>
            <div
              className="align-left UcastniciTable_oddRow UcastniciTable_cell"
              style={
                Object {
                  "height": 35,
                  "left": 100,
                  "position": "absolute",
                  "top": 0,
                  "width": 90,
                }
              }
            >
              Roman
            </div>
            <div
              className="align-right UcastniciTable_oddRow UcastniciTable_cell"
              style={
                Object {
                  "height": 35,
                  "left": 190,
                  "position": "absolute",
                  "top": 0,
                  "width": 100,
                }
              }
            >
              1956
            </div>
            <div
              className="align-left UcastniciTable_evenRow UcastniciTable_cell"
              style={
                Object {
                  "height": 35,
                  "left": 0,
                  "position": "absolute",
                  "top": 35,
                  "width": 100,
                }
              }
            >
              <a
                href="/prihlasky/6f09b1fd371dec1e99b7e1c9"
                onClick={[Function]}
              >
                Sukdoláková
              </a>
            </div>
            <div
              className="align-left UcastniciTable_evenRow UcastniciTable_cell"
              style={
                Object {
                  "height": 35,
                  "left": 100,
                  "position": "absolute",
                  "top": 35,
                  "width": 90,
                }
              }
            >
              Martina
            </div>
            <div
              className="align-right UcastniciTable_evenRow UcastniciTable_cell"
              style={
                Object {
                  "height": 35,
                  "left": 190,
                  "position": "absolute",
                  "top": 35,
                  "width": 100,
                }
              }
            >
              7. 12. 1963
            </div>
            <div
              className="align-left UcastniciTable_oddRow UcastniciTable_cell"
              style={
                Object {
                  "height": 35,
                  "left": 0,
                  "position": "absolute",
                  "top": 70,
                  "width": 100,
                }
              }
            >
              <a
                href="/prihlasky/7a09b1fd371dec1e99b7e142"
                onClick={[Function]}
              >
                Zralá
              </a>
            </div>
            <div
              className="align-left UcastniciTable_oddRow UcastniciTable_cell"
              style={
                Object {
                  "height": 35,
                  "left": 100,
                  "position": "absolute",
                  "top": 70,
                  "width": 90,
                }
              }
            >
              Hana
            </div>
            <div
              className="align-right UcastniciTable_oddRow UcastniciTable_cell"
              style={
                Object {
                  "height": 35,
                  "left": 190,
                  "position": "absolute",
                  "top": 70,
                  "width": 100,
                }
              }
            >
              25. 7. 1999
            </div>
          </div>
        </div>
      </div>
      <div
        className="UcastniciTable_GridColumn"
      >
        <div>
          <div
            style={
              Object {
                "height": 35,
                "width": 0,
              }
            }
          >
            <div
              aria-label="grid"
              aria-readonly={true}
              className="ReactVirtualized__Grid UcastniciTable_HeaderGrid"
              onScroll={[Function]}
              role="grid"
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "boxSizing": "border-box",
                  "direction": "ltr",
                  "height": 35,
                  "overflowX": "hidden",
                  "overflowY": "hidden",
                  "position": "relative",
                  "width": 290,
                  "willChange": "transform",
                }
              }
              tabIndex={0}
            />
          </div>
          <div
            style={
              Object {
                "height": 105,
                "width": 290,
              }
            }
          >
            <div
              aria-label="grid"
              aria-readonly={true}
              className="ReactVirtualized__Grid UcastniciTable_BodyGrid"
              onScroll={[Function]}
              role="grid"
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "boxSizing": "border-box",
                  "direction": "ltr",
                  "height": 105,
                  "overflowX": "hidden",
                  "overflowY": "hidden",
                  "position": "relative",
                  "width": 290,
                  "willChange": "transform",
                }
              }
              tabIndex={0}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`žádný pohár 1`] = `
<div
  className="Pohar__div UcastniciTable_container"
>
  <div>
    <input
      className="TextFilter__input"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Filtr na příjmení a jméno"
      type="text"
      value=""
    />
    <span
      className="Pohar__buttons Bootstrap-buttons--active"
    >
      <button
        className="Pohar__button btn btn-success"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          className="glyphicon glyphicon-download"
        />
         Nárok na pohár
      </button>
      <button
        className="Pohar__button btn btn-info"
        disabled={false}
        onClick={[MockFunction]}
        type="button"
      >
        <span
          className="glyphicon glyphicon-upload"
        />
         Nepřevzatý pohár
      </button>
    </span>
    <span
      className="Zobrazeno"
    >
      zobrazeno: 
      <span
        className="badge"
      >
        0
      </span>
    </span>
  </div>
  <div
    className="withResponsive"
  >
    <div
      className="UcastniciTable_GridRow"
    >
      <div
        className="UcastniciTable_LeftSideGridContainer"
        style={
          Object {
            "flexBasis": "290px",
            "top": 0,
          }
        }
      >
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid UcastniciTable_HeaderGrid"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 35,
              "overflowX": "hidden",
              "overflowY": "hidden",
              "position": "relative",
              "width": 290,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        >
          <div
            className="ReactVirtualized__Grid__innerScrollContainer"
            role="rowgroup"
            style={
              Object {
                "height": 35,
                "maxHeight": 35,
                "maxWidth": 290,
                "overflow": "hidden",
                "pointerEvents": "",
                "position": "relative",
                "width": 290,
              }
            }
          >
            <div
              className="UcastniciTable_headerCell"
              style={
                Object {
                  "height": 35,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 100,
                }
              }
            >
              <button
                className="SortHeader"
                onClick={[Function]}
              >
                příjmení
                 
                
              </button>
            </div>
            <div
              className="UcastniciTable_headerCell"
              style={
                Object {
                  "height": 35,
                  "left": 100,
                  "position": "absolute",
                  "top": 0,
                  "width": 90,
                }
              }
            >
              <button
                className="SortHeader"
                onClick={[Function]}
              >
                jméno
                 
                
              </button>
            </div>
            <div
              className="UcastniciTable_headerCell"
              style={
                Object {
                  "height": 35,
                  "left": 190,
                  "position": "absolute",
                  "top": 0,
                  "width": 100,
                }
              }
            >
              <button
                className="SortHeader"
                onClick={[Function]}
              >
                narození
                 
                
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="UcastniciTable_LeftSideGridContainer"
        style={
          Object {
            "flexBasis": "290px",
            "top": 35,
          }
        }
      >
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid UcastniciTable_LeftSideGrid"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 0,
              "overflowX": "auto",
              "overflowY": "hidden",
              "position": "relative",
              "width": 290,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        />
      </div>
      <div
        className="UcastniciTable_GridColumn"
      >
        <div>
          <div
            style={
              Object {
                "height": 35,
                "width": 0,
              }
            }
          >
            <div
              aria-label="grid"
              aria-readonly={true}
              className="ReactVirtualized__Grid UcastniciTable_HeaderGrid"
              onScroll={[Function]}
              role="grid"
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "boxSizing": "border-box",
                  "direction": "ltr",
                  "height": 35,
                  "overflowX": "hidden",
                  "overflowY": "hidden",
                  "position": "relative",
                  "width": 290,
                  "willChange": "transform",
                }
              }
              tabIndex={0}
            />
          </div>
          <div
            style={
              Object {
                "height": 0,
                "width": 290,
              }
            }
          >
            <div
              aria-label="grid"
              aria-readonly={true}
              className="ReactVirtualized__Grid UcastniciTable_BodyGrid"
              onScroll={[Function]}
              role="grid"
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "boxSizing": "border-box",
                  "direction": "ltr",
                  "height": 0,
                  "overflowX": "auto",
                  "overflowY": "hidden",
                  "position": "relative",
                  "width": 290,
                  "willChange": "transform",
                }
              }
              tabIndex={0}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
